### DISCUSSION TITLE
Unreal 5.0 question

### DISCUSSION THREAD

**Discussion by MrTeaRexx** [Sixze/ALS-Refactored#71] [General]
with 5.0 released will we see a update of the old ue4 skeleton? cause the UE5 version is no longer using it and was curious if will see the same upgrade to ALS?

**Comment by Sixze** [comment]
Unreal Engine 5.0 introduced the **IK Retargeter** and something called **Skeleton Compatibility**, which should make it easier to retarget existing animations to the new skeleton.

As for the complete transition to a new skeleton, I can't promise much. All existing animations and ALS mannequin mesh need to be reworked, which I probably won't be able to do on my own.

**Reply by kriNon** [reply]
Have you looked into the **runtime retargeting** in the new IK retargeting system?
Might be possible to keep using the existing skeleton, but make it so that people can runtime retarget to their own choice of skeleton provided they set up their own IK rigs and IK Retargeter.

There's a good example of this in the UE5 version of the **content examples** project.
The retargeting is set up here: AnimBlueprint'/Game/ExampleContent/AnimationRetargeting/AnimBlueprints/ABP_Echo_Retargeting.ABP_Echo_Retargeting'
![image](https://user-images.githubusercontent.com/3295424/163330447-f906e648-2f58-4208-9af5-df79fd8cfcd4.png)

The result can be viewed in 1.2 of the **Animation Retargeting** level.

**Reply by Sixze** [reply]
Not yet, but I'm going to look into this in the future. I also plan to add a new mannequin from Lyra Game as an example of new retargeting features.

**Reply by gswec2005** [reply]
Hello, friends!

Newba here trying to learn with you all <3

For now, i'm testing using a child skeletal mesh on ALS Character and put a transparent material on als character with 0 opacity.
![image](https://user-images.githubusercontent.com/13799808/163453201-d2bbbe71-6b11-4c65-abbc-932bf22c1761.png)
![image](https://user-images.githubusercontent.com/13799808/163453321-42b48b65-4bf4-4b5a-a4e7-74541391e962.png)
![image](https://user-images.githubusercontent.com/13799808/163453402-256df826-ff7d-482d-a6e1-678c1f019411.png)


This SkMesh has SKM_MannySimple, and ABP_Manny_C as animblueprint modified
![image](https://user-images.githubusercontent.com/13799808/163453589-a246d671-c2f5-46bc-b4d4-d94b169971f5.png)


here it working like a charm, i've heard of fps issues about this method, and see some bone bugs, but for start my learning, i loved it <3
![image](https://user-images.githubusercontent.com/13799808/163454108-b2434cad-699f-4c77-a0c1-893484c39b07.png)
![image](https://user-images.githubusercontent.com/13799808/163454157-fe6e7efd-1458-4b1a-97a6-df7c23a6ceba.png)

**Reply by kriNon** [reply]
I think that removing the mesh from the ALS character mesh should perform better than making it transparent. I think you should be able to take it into blender and delete the mesh but keep the skeleton untouched.

As for fps issues, the runtime retargeting should actually be pretty performant. If I understand correctly, the retargeting is performed in a separate thread from your game logic.
